ui/src/admin/containers/influxdb/RolePage.tsx

```diff
@@ -0,0 +1,407 @@
+  const [allUserNames, usersRecord] = useMemo(() => {
+    const uNames = users.map(r => r.name).sort()
+    const ruRecord = role.users.reduce<Record<string, boolean>>((acc, r) => {
+      acc[r.name] = true
+      return acc
+    }, {})
+    return [uNames, ruRecord]
+  }, [role, users])
+  const [changedUsersRecord, setChangedUsersRecord] = useState<
+    Record<string, boolean>
+  >({})
+  useEffect(() => setChangedUsersRecord({}), [role])
+  const usersChanged = useMemo(() => !!Object.keys(changedUsersRecord).length, [
+    changedUsersRecord,
+  ])
+  const onUserChange = useMemo(
+    () => (e: React.MouseEvent<HTMLElement>) => {
+      const user = (e.target as HTMLElement).dataset.user
+      const {[user]: roleChanged, ...restChanged} = changedUsersRecord
+      if (roleChanged === undefined) {
+        setChangedUsersRecord({
+          ...changedUsersRecord,
+          [user]: !usersRecord[user],
+        })
+      } else {
+        // returning back to original state
+        setChangedUsersRecord(restChanged)
+      }
+      return
+    },
+    [usersRecord, changedUsersRecord]
+  )
+  const changeUsers = useMemo(
+    () => async () => {
+      if (Object.entries(changedUsersRecord).length === 0) {
+        return
+      }
+      setRunning(true)
+      try {
+        const newUsers = users.reduce<User[]>((acc, user) => {
+          const userName = user.name
+          if (
+            changedUsersRecord[userName] === true ||
+            (changedUsersRecord[userName] === undefined &&
+              usersRecord[userName])
+          ) {
+            acc.push(user)
+          }
+          return acc
+        }, [])
+        await updateUsersAsync(role, newUsers)
+      } finally {
+        setRunning(false)
+      }
+    },
+    [role, usersRecord, changedUsersRecord, users]
+  )
+
+  const dataChanged = useMemo(() => permissionsChanged || usersChanged, [
+    permissionsChanged,
+    usersChanged,
+  ])
+  const changeData = useCallback(async () => {
+    await changeUsers()
+    await changePermissions()
+  }, [changePermissions, changeUsers])
+  const exitHandler = useCallback(() => {
+    router.push(`/sources/${sourceID}/admin-influxdb/roles`)
+  }, [router, source])
+  const databaseNames = useMemo<string[]>(
+    () =>
+      databases.reduce(
+        (acc, db) => {
+          acc.push(db.name)
+          return acc
+        },
+        ['']
+      ),
+    [databases]
+  )
+  const body =
+    role === FAKE_ROLE ? (
+      <div className="container-fluid">
+        User <span className="error-warning">{roleName}</span> not found!
+      </div>
+    ) : (
+      <div className="panel panel-solid influxdb-admin">
+        <div className="panel-heading">
+          <h2 className="panel-title">
+            <span title={`Role: ${roleName}`}>{roleName}</span>
+          </h2>
+          <div className="panel-heading--right">
+            <ConfirmButton
+              type="btn-danger"
+              text="Delete User"
```

Comment:
The button should say `Delete Role`, as you called it in the text.
```suggestion
              text="Delete Role"
```